app/security.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"runtime"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+)
+
+const (
+	SECURITY_URL = "https://d7zmvsa9e04kk.cloudfront.net"
+
+	PROP_SECURITY_ID                = "id"
+	PROP_SECURITY_CATEGORY          = "c"
+	VAL_SECURITY_CATEGORY_DEFAULT   = "d"
+	PROP_SECURITY_BUILD             = "b"
+	PROP_SECURITY_ENTERPRISE_READY  = "be"
+	PROP_SECURITY_DATABASE          = "db"
+	PROP_SECURITY_OS                = "os"
+	PROP_SECURITY_USER_COUNT        = "uc"
+	PROP_SECURITY_TEAM_COUNT        = "tc"
+	PROP_SECURITY_ACTIVE_USER_COUNT = "auc"
+	PROP_SECURITY_UNIT_TESTS        = "ut"
+)
+
+func DoSecurityUpdateCheck() {
+	if *utils.Cfg.ServiceSettings.EnableSecurityFixAlert {
+		if result := <-Srv.Store.System().Get(); result.Err == nil {
+			props := result.Data.(model.StringMap)
+			lastSecurityTime, _ := strconv.ParseInt(props[model.SYSTEM_LAST_SECURITY_TIME], 10, 0)
+			currentTime := model.GetMillis()
+
+			if (currentTime - lastSecurityTime) > 1000*60*60*24*1 {
+				l4g.Debug(utils.T("mattermost.security_checks.debug"))
+
+				v := url.Values{}
```

Comment:
Why do we still send a bunch of stuff with the security check if we are sending it to segment as well?